src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -83,20 +83,33 @@ internal static partial class DependentTypeFinder
             // are passed in.  There is no need to check D as there's no way it could
             // contribute an intermediate type that affects A or C.  We only need to check
             // A, B and C
+            //
+            // An exception to the above rule is if we're just searching a single project.
+            // in that case there can be no intermediary projects that could add types.
+            // So we can just limit ourselves to that single project.
 
             // First find all the projects that could potentially reference this type.
-            var projectsThatCouldReferenceType = await GetProjectsThatCouldReferenceTypeAsync(
-                type, solution, searchInMetadata, cancellationToken).ConfigureAwait(false);
+            List<Project> orderedProjectsToExamine;
```

Comment:
this is a good question.  The reason is more philosophical than anything else.  I'm generally wary of IEnumerables as it can be very hard to gauge certain things (for example: how expensive htey are, or if they return teh same results when querying them multiple times).  A list is a well known construct that i can be sure is computed and will give me consistent results.